kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Scheduler.scala

```diff
@@ -74,55 +68,54 @@ final class Scheduler(
         ExecutionContext.fromExecutor(asExecutor)
 
     private def schedule(task: Task, submitter: Worker): Unit = {
-        val cycles = this.cycles
-        @tailrec def loop(tries: Int = 0): Unit = {
-            var worker: Worker = null
-            if (submitter == null && tries == 0)
-                worker = Worker.current()
-            if (worker == null) {
-                val currentWorkers = this.currentWorkers
-                var position       = XSRandom.nextInt(currentWorkers)
-                var stride         = Math.min(currentWorkers, scheduleStride)
-                var minLoad        = Int.MaxValue
-                while (stride > 0 && minLoad != 0) {
-                    val candidate = workers(position)
-                    if (
-                        candidate != null &&
-                        (candidate ne submitter) &&
-                        candidate.checkAvailability(cycles)
-                    ) {
-                        val l = candidate.load()
-                        if (l < minLoad) {
-                            minLoad = l
-                            worker = candidate
-                        }
+        val nowMs          = clock.currentMillis()
+        var worker: Worker = null
+        if (submitter == null) {
+            worker = Worker.current()
+            if (worker != null && !worker.checkAvailability(nowMs))
```

Comment:
This check was missing. If the current worker is not available, the task shouldn't be enqueued locally.

Reply:
Won't this drain the worker? Should we have a `checkAvailability` which doesn't drain?